### PR TITLE
Hotfix/walk to pokestop and refactor to walker

### PR DESCRIPTION
**Pull request by douglascamata**

### CODE REVIEW

**Comment by PP19** [comment]
Bot just worked 5 minutes using this pull request, and then did nothing

[14:57:02] Walking from (48.19526927058824, 16.337285505882353) to [48.1957363, 16.3368393] 0.06km
^C[15:00:05] Exiting PokemonGo Bot

**Comment by Sfim** [comment]
[15:03:08] Scanning area for objects....
[15:03:08] Walking from (49.1151552, -1.0946844) to [49.1151552, -1.0946844] 0.00km
[========================================] 100%
[15:03:12] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.04km
[15:03:12] Need to move closer to Pokestop
[                                        ] 0%
[15:03:15] Scanning area for objects....
[15:03:15] Walking from (49.11512291436413, -1.0946849352514783) to [49.1169552, -1.0946844] 0.20km
[                                        ] 0%
[15:03:23] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.04km
[15:03:23] Need to move closer to Pokestop
[                                        ] 0%
[15:03:35] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.04km
[15:03:35] Need to move closer to Pokestop
[                                        ] 0%
[15:03:49] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.04km
[15:03:49] Need to move closer to Pokestop
[                                        ] 0%
[15:04:00] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.05km
[15:04:00] Need to move closer to Pokestop
[                                        ] 0%
[15:04:12] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.05km
[15:04:12] Need to move closer to Pokestop
[                                        ] 1%
[15:04:22] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.05km
[15:04:22] Need to move closer to Pokestop
[                                        ] 2%
[15:04:33] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.05km
[15:04:33] Need to move closer to Pokestop
[                                        ] 1%
[15:04:43] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.05km
[15:04:43] Need to move closer to Pokestop
[                                        ] 1%
[15:04:54] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.05km
[15:04:54] Need to move closer to Pokestop
[                                        ] 0%
[15:05:06] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.04km
[15:05:06] Need to move closer to Pokestop
[                                        ] 0%
[15:05:18] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.04km
[15:05:18] Need to move closer to Pokestop
[                                        ] 0%
[15:05:29] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.04km
!

[path bug](https://cloud.githubusercontent.com/assets/20683278/17176327/65b27412-540c-11e6-8675-4b046350eabe.png)

**Comment by mrseeker** [comment]
That path bug seems to be also in dev...

**Comment by douglascamata** [comment]
test again, guys.... I pushed a probable fix.

**Comment by mrseeker** [comment]
Can confirm, last commit did fix the path bug:
```
[09:22:17] [x] Found fort aee9ef1060e049809afb11755ed0a52b.16 at distance 0.03km
[09:22:17] [x] Need to move closer to Pokestop
[09:22:22] [x] Found fort aee9ef1060e049809afb11755ed0a52b.16 at distance 0.02km
[09:22:22] [x] Need to move closer to Pokestop
[09:22:28] [x] Found fort aee9ef1060e049809afb11755ed0a52b.16 at distance 0.02km
[09:22:28] [x] Need to move closer to Pokestop
[09:22:39] [x] Found fort aee9ef1060e049809afb11755ed0a52b.16 at distance 0.01km
[09:22:39] [x] Need to move closer to Pokestop
[09:22:44] [x] Found fort aee9ef1060e049809afb11755ed0a52b.16 at distance 0.01km
[09:22:44] [o] Arrived at Pokestop
[09:22:45] Now at Pokestop: Het blauwe theehuis - Spinning...
```

**Comment by Sfim** [comment]
sorry but not seem ok 
code refresh done

Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.04km
[15:25:07] Need to move closer to Pokestop
[                                        ] 0%
[15:25:18] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.04km
[15:25:18] Need to move closer to Pokestop
[                                        ] 0%
[15:25:30] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.04km
[15:25:30] Need to move closer to Pokestop
[                                        ] 0%
[15:25:41] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.04km
[15:25:41] Need to move closer to Pokestop
[                                        ] 0%
[15:25:53] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.04km
[15:25:53] Need to move closer to Pokestop
[                                        ] 0%

path bug 2

**Comment by tstumm** [comment]
@Sfim How did you apply the PR? Just pulling `dev` is not enough.

**Comment by mrseeker** [comment]
@Sfim Check if you are really using the last commit, and use the command `git checkout hotfix/walk-to-pokestop-and-refactor-to-walker` since this is not on dev but on its own branch.

**Comment by drantor** [comment]
Looks very good with the patch :+1:

bildschirmfoto 2016-07-27 um 15 32 09

**Comment by tstumm** [comment]
I'd say it working fair enough.
👍

**Comment by PP19** [comment]
Using the latest commits, the bot is now crashing with 



`[15:32:16] [x] Found fort dafdcda146ee43728a3d7f..... at distance 0.01km
[15:32:16] [x] Need to move closer to Pokestop
Traceback (most recent call last):
  File "pokecli.py", line 283, in <module>
    main()
  File "pokecli.py", line 249, in main
    bot.take_step()
  File "/home/user/pgobot/PokemonGo-Bot/pokemongo_bot/__init__.py", line 46, in take_step
    self.process_cells(work_on_forts=True)
  File "/home/user/pgobot/PokemonGo-Bot/pokemongo_bot/__init__.py", line 66, in process_cells
    "catchable_pokemons": catchable_pokemons}, location, work_on_forts)
  File "/home/user/pgobot/PokemonGo-Bot/pokemongo_bot/__init__.py", line 170, in work_on_cell
    RecycleItemsWorker(self).work()
  File "/home/user/pgobot/PokemonGo-Bot/pokemongo_bot/cell_workers/recycle_items_worker.py", line 21, in work
    bag_count = self.bot.item_inventory_count(item_id)
  File "/home/user/pgobot/PokemonGo-Bot/pokemongo_bot/__init__.py", line 377, in item_inventory_count
    inventory_dict = inventory_req['responses'][
TypeError: 'bool' object has no attribute '__getitem__'
`

**Comment by lukaszraczylo** [comment]
Works awesome ( after merging with dev branch, just in case :) ) Great stuff.

**Comment by Sfim** [comment]
my bad it's was just dig pull
this time it's ok for me to 

[15:33:54] [x] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.04km
[15:33:54] [x] Need to move closer to Pokestop
[15:34:01] [x] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.04km
[15:34:01] [x] Need to move closer to Pokestop
[15:34:07] [x] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.04km
[15:34:07] [x] Need to move closer to Pokestop
[15:34:13] [x] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.03km
[15:34:13] [x] Need to move closer to Pokestop
[15:34:20] [x] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.03km
[15:34:20] [x] Need to move closer to Pokestop
[15:34:28] [x] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.02km
[15:34:28] [x] Need to move closer to Pokestop
[15:34:35] [x] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.02km
[15:34:35] [x] Need to move closer to Pokestop
[15:34:41] [x] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.01km
[15:34:41] [x] Need to move closer to Pokestop
[15:34:48] [x] Found fort fa993a77336d408b9dc08746b19a9416.16 at distance 0.01km

**Comment by douglascamata** [comment]
@PP19 servers are under heavy load and giving faulty responses in the last few hours...

**Comment by Sfim** [comment]
i don't know if it's the modification of the path script but on the dev version we meet few pokemon instead of the master version on the same location

**Comment by douglascamata** [comment]
@Sfim 

> @PP19 servers are under heavy load and giving faulty responses in the last few hours...

File: pokemongo_bot/cell_workers/evolve_all_worker.py [hunk 1]
```diff
@@ -109,7 +109,7 @@ def _release_evolved(self, release_cand_list_ids):
                     # Transfering Pokemon
                     self.transfer_pokemon(pokemon_id)
                     logger.log(
-                        '[#] {} has been exchanged for candy!'.format(pokemon_name), 'green')
+                        '[#] {} has been exchanged for candy!'.format(pokemon_name), 'red')
```
Comment (MaxLeiter) [inline pokemongo_bot/cell_workers/evolve_all_worker.py]: Why this change? Red means bad, green means positive, this shouldn't be changed.

Comment (douglascamata) [inline pokemongo_bot/cell_workers/evolve_all_worker.py]: Red means `pay attention`.

**Comment by T3hKonijn** [comment]
>>completely removed walking progress bar (log trimming is required)

Can we have this back with a config option or something?
